app/controllers/katello/concerns/api/v2/registration_controller_extensions.rb

```diff
@@ -5,7 +5,8 @@ module Api::V2::RegistrationControllerExtensions
 
       def prepare_host
         if params['uuid']
-          @host = Katello::Host::SubscriptionFacet.find_by(uuid: params['uuid']).host
+          @host = Katello::Host::SubscriptionFacet.find_by(uuid: params['uuid'])&.host
+          fail ActiveRecord::RecordNotFound, N_('Host not found') if @host.nil?
```

Comment:
I think the fix is right but it would be good to explain what happend in the exception message. Could you please change it to "Host was not found by the subscription UUID: '%s', this can happen if the host is registered already, but not to this Foreman"